packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.d.ts

```diff
@@ -77,14 +68,101 @@ export type InlineContentStyle = {
   justify: JustifyContent;
 };
 
-export default class InlineContent extends Base {
+declare namespace InlineContent {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * an array of  data to render inline
+     */
+    content?: TextContent[];
+    /**
+     * object setting flexItem props on all content items
+     */
+    contentProperties?: FlexItem;
+    /**
+     * y value of all badges
+     */
+    badgeY?: number;
+    /**
+     * styling to update the Badge component
+     */
+    badgeProperties?: Partial<Badge>;
+    /**
+     * alignment of first line flexbox content
+     */
+    justify?: JustifyContent;
+    /**
+     * determines whether the containing flexbox should wrap the content onto the next line
+     */
+    contentWrap?: boolean;
+    /**
+     * set of nested objects containing [Lightning Text](https://rdkcentral.github.io/Lightning/docs/textures/text) properties which can be referenced in the [Content Element Properties](#content-element-properties) to apply styling to individual text elements
+     */
+    customStyleMappings?: Record<string, unknown>;
```

Comment:
I don't believe `Record<string, unknown>` will allow for nested objects. if the expected object is something like:
```
{
  elementName: {
    customStyle: value,
    customStyle2: value
  }
}
```

something like this may work better:
```
Record<string, Record<string, unknown>>
```

 if there would be more nesting beyond that, then:
```
Record<string, Record<string, any>>
``` 
as much as I'd prefer to avoid any